File: frontend/authorization_server_website/src/app/models/ManagerResponse.ts
import {RoleTemplate} from './RoleTemplate';

export class ManagerResponse {
  constructor() {
  }

  private _ownerId: number;

  get ownerId(): number {
    return this._ownerId;
  }

  set ownerId(value: number) {
    this._ownerId = value;
  }

  private _clientId: string;

  get clientId(): string {
    return this._clientId;
  }

  set clientId(value: string) {
    this._clientId = value;
  }

  private _clientName: string;

  get clientName(): string {
    return this._clientName;
  }

  set clientName(value: string) {
    this._clientName = value;
  }

  private _rolesTemplates: RoleTemplate[];

  get rolesTemplates(): RoleTemplate[] {
    return this._rolesTemplates;
  }

  set rolesTemplates(value: RoleTemplate[]) {
    this._rolesTemplates = value;
  }
}

File: backend/authorization_server/src/main/java/com/example/pkce/models/ManagerResponse.java
package com.example.pkce.models;

import com.example.pkce.entities.Client;
import com.example.pkce.entities.RoleTemplate;

import java.util.List;

public class ManagerResponse {

    private int ownerId;
    private String clientId;
    private String clientName;
    private List<RoleTemplate> rolesTemplates;

    public ManagerResponse() {
    }

    public ManagerResponse(Client client){
        this.ownerId = client.getOwner();
        this.clientId = client.getClientId();
        this.clientName = client.getName();
    }

    public int getOwnerId() {
        return ownerId;
    }

    public void setOwnerId(int ownerId) {
        this.ownerId = ownerId;
    }

    public String getClientId() {
        return clientId;
    }

    public void setClientId(String clientId) {
        this.clientId = clientId;
    }

    public String getClientName() {
        return clientName;
    }

    public void setClientName(String clientName) {
        this.clientName = clientName;
    }

    public List<RoleTemplate> getRolesTemplates() {
        return rolesTemplates;
    }

    public void setRolesTemplates(List<RoleTemplate> rolesTemplates) {
        this.rolesTemplates = rolesTemplates;
    }
}
